HIP/hip-655.md

```diff
@@ -0,0 +1,106 @@
+
+### New APIs/UIs:
+
+- Services:
+    - `AcceptUnsolicitedToken` API
+- Mirror node:
+    - GET `unsolicitedInbox` query for the Airdrop-Recipient to get content of their unsolicited-inbox.
+    - GET `unsolicitedOutbox` query for the Airdrop-Sender to get content of their unsolicited-outbox.
+- Wallets and Explorers
+    - Wallets and Explorers should provide a way for users to be able to see the unsolicited-inbox and accept the tokens that they wish to accept
+
+### Other considerations
+
+- Auto-association: if the recipient of the token has *open* auto-association slots, then none of the mechanisms described in this document apply. That is treated as a regular transfer. The airdropped token will be auto-associated with the recipient and the transfer will succeed as it does today
+- Already associated token type: if the airdropped token type is already associated with the recipient’s account, then none of the mechanisms described in this document apply. That is treated as a regular transfer. The airdropped token will simply be transferred to the recipient's account as it does today.
+- Receiver-signature-required: if the Airdrop-Recipient has receiver-signature-required set on their account, then the airdrop will only succeed if they also sign the `AcceptUnsolicitedToken` API with their receiver-signature.
+- Custom fees on the token: The Airdrop-Sender will pay for the custom-fees levied on the token transfer at the time of the transfer (i.e. the time at which the Airdrop-Receiver signed the `AcceptUnsolicitedToken` transaction). Note that the custom-fees may change from the time the Airdrop-sender attempted to send the tokens to the time the actual transfer takes place, and it is a risk that the Airdrop-Sender will have to assume.
+- Token deletion: if the token is deleted before the Airdrop-Receiver accepts the token, the `AcceptUnsolicitedToken` call will simply fail with an appropriate error message. This proposal is compatible with the concept of token associations as implemented on the network today. It builds on that concept but does not violate any of the existing concepts of token associations. Therefore, existing applications will not be adversely affected due to the implementation of this HIP.
+- Information to the user: With this mechanism described in this document, the Airdrop-Sender will have to check their unsolicited-outbox to see which tokens have not been deposited to the Airdrop-Recipients’ accounts; and the airdrop-Recipient will have to check their unsolicited-inbox to see which tokens are pending their acceptance.
+- Wallets, Exchanges, and mirror-node explorers will have to create visibility and notification mechanisms for the users.
+        
+## Backward Compatibility
```

Comment:
If response code 22 `SUCCESS` is not returned from a smart contract precompile `cryptoTransfer`, some widely used smart contracts on the Hedera network will not work as intended. For example, in order to "airdrop" to yourself a new token by swapping an associated token for it on a DEX, response code 22 is required.

Reply:
This is a really good point. 
A transfer needs to be able to convey back the interim period of outbox status to not give the wrong impression.
🤔